store/store.go

```diff
@@ -178,13 +178,24 @@ func (s *Store[H]) Height() uint64 {
 	return s.heightSub.Height()
 }
 
-func (s *Store[H]) Head(_ context.Context, _ ...header.HeadOption[H]) (H, error) {
-	if head := s.contiguousHead.Load(); head != nil {
-		return *head, nil
+func (s *Store[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, error) {
+	head, err := s.GetByHeight(ctx, s.heightSub.Height())
```

Comment:
You can also try check loading contiguousHead here and if nil, then try readHead

Reply:
Done